src/files/normalise-input.js

```diff
@@ -0,0 +1,233 @@
+'use strict'
+
+const errCode = require('err-code')
+const { Buffer } = require('buffer')
+
+/*
+ * Transform one of:
+ *
+ * ```
+ * Buffer|ArrayBuffer|TypedArray
+ * Blob|File
+ * { path, content: Blob }
+ * { path, content: String }
+ * { path, content: Iterable<Number> }
+ * { path, content: Iterable<Buffer> }
+ * { path, content: Iterable<Iterable<Number>> }
+ * { path, content: AsyncIterable<Iterable<Number>> }
+ * String
+ * Iterable<Number>
+ * Iterable<Buffer>
+ * Iterable<Blob>
+ * Iterable<{ path, content: Buffer }>
+ * Iterable<{ path, content: Blob }>
+ * Iterable<{ path, content: Iterable<Number> }>
+ * Iterable<{ path, content: AsyncIterable<Buffer> }>
+ * AsyncIterable<Buffer>
+ * AsyncIterable<{ path, content: Buffer }>
+ * AsyncIterable<{ path, content: Blob }>
+ * AsyncIterable<{ path, content: Iterable<Buffer> }>
+ * AsyncIterable<{ path, content: AsyncIterable<Buffer> }>
+ * ```
+ * Into:
+ *
+ * ```
+ * AsyncIterable<{ path, content: AsyncIterable<Buffer> }>
+ * ```
+ *
+ * @param input Object
+ * @return AsyncInterable<{ path, content: AsyncIterable<Buffer> }>
+ */
+module.exports = function normaliseInput (input) {
+  // must give us something
+  if (input === null || input === undefined) {
+    throw errCode(new Error(`Unexpected input: ${input}`, 'ERR_UNEXPECTED_INPUT'))
+  }
+
+  // { path, content: ? }
+  if (isFileObject(input)) {
+    return (async function * () { // eslint-disable-line require-await
+      yield toFileObject(input)
+    })()
+  }
+
+  // String
+  if (typeof input === 'string' || input instanceof String) {
+    return (async function * () { // eslint-disable-line require-await
+      yield toFileObject(input)
+    })()
+  }
+
+  // Buffer|ArrayBuffer|TypedArray
+  // Blob|File
+  if (isBytes(input) || isBloby(input)) {
+    return (async function * () { // eslint-disable-line require-await
+      yield toFileObject(input)
+    })()
+  }
+
+  // Iterable<?>
+  if (input[Symbol.iterator]) {
+    // Iterable<Number>
+    if (!isNaN(input[0])) {
+      return (async function * () { // eslint-disable-line require-await
+        yield toFileObject([input])
+      })()
+    }
+
+    // Iterable<Buffer>
```

Comment:
It's werid that `Iterable<Buffer>` means multiple files but `AsyncIterable<Buffer>` is a single file...

As I mentioned in an earlier comment I reckon we should interpret sets of `Bytes` as a single file and sets of `Blobys` as multiple files.

Reply:
Yeah - multiple Buffers should be treated as one file, multiple Blobs are multiple files.  That's the intention anyway, I guess the comment isn't clear.

Realistically you'd need `{ path, content }` objects for multiple Blobs as they have no path themselves and the importer will fail complaining of multiple roots.

Alternatively we could widen it to accept HTML File objects which have a `.name` property and use that as the path?  The user would still need to pass the `wrapWithDirectory` arg to the importer though because I think the `.name` property is supposed to just be the filename, not the path to the file within a directory structure as with the `{ path, content }` object.

Urgh, what fun.